autogpt_platform/frontend/src/components/agptui/LibraryNotificationDropdown.tsx

```diff
@@ -0,0 +1,130 @@
+"use client";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuLabel,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { Button } from "./Button";
+import { BellIcon, X } from "lucide-react";
+import { motion, useAnimationControls } from "framer-motion";
+import { useState, useEffect, useMemo } from "react";
+import LibraryNotificationCard, {
+  NotificationCardData,
+} from "./LibraryNotificationCard";
+import { cn } from "@/lib/utils";
+
+const LibraryNotificationDropdown = () => {
+  const controls = useAnimationControls();
+  const [open, setOpen] = useState(false);
+  const [notifications, setNotifications] = useState<
+    NotificationCardData[] | null
+  >(null);
+
+  const initialNotificationData = useMemo(
```

Comment:
should this live here or should we have skeletons?

Reply:
- #9515